unidist/core/backends/mpi/core/communication.py

```diff
@@ -132,9 +132,16 @@ def __init__(self, comm):
             self.topology[host][host_rank] = global_rank
             self.host_by_rank[global_rank] = host
 
-        self.monitor_processes = [
-            self.topology[host][MPIRank.MONITOR] for host in self.topology
-        ]
+        if common.is_shared_memory_supported():
+            self.monitor_processes = []
+            for host in self.topology:
+                if len(self.topology[host]) < 2:
+                    raise ValueError(
+                        "When using shared memory, each host must contain at least 2 processes, since one of them will be a service monitor."
```

Comment:
```suggestion
                        "When using shared object store, each host must contain at least 2 processes, "
                        "since one of them will be a service monitor."
```

Reply:
done